src/v/raft/recovery_throttle.h

```diff
@@ -100,13 +107,36 @@ class recovery_throttle {
         refresh();
         /*
          * if a waiter exists continue refreshing since it is not guaranteed
-         * that throttle will be invoked (e.g. excactly one recovering group).
+         * that throttle will be invoked (e.g. exactly one recovering group).
          */
         if (_sem.waiters()) {
             _refresh_timer.arm(refresh_interval);
         }
     }
 
+    size_t get_per_core_rate() { return _rate_binding() / ss::smp::count; }
+
+    void update_rate() {
+        auto const per_core_rate = get_per_core_rate();
+
+        if (_rate == per_core_rate) {
+            return;
+        }
+
+        vlog(
+          raftlog.info,
+          "Updating recovery throttle with new rate of {}",
+          per_core_rate);
+
+        if (_rate > per_core_rate) {
+            _sem.consume(_rate - per_core_rate);
```

Comment:
q: Do we need to define min/max for this config? Think we can lock this up if we change it to a small value by mistake.

Reply:
that is true, i will add max/min pair